workflows/metaval.nf

```diff
@@ -85,15 +92,54 @@ workflow METAVAL {
         ch_extract_reads.diamond_tsv,
         )
         ch_versions            = ch_versions.mix( TAXID_READS.out.versions )
-    }
-
-    /*
-        SUBWORKFLOW: Screen pathogens
-    */
-    ch_reference = Channel.fromPath( params.pathogens_genome, checkIfExists: true)
+        // Remove empty fastq files produced by extracting reads for user defined taxIDs
+        if (params.extract_kraken2_reads && params.taxid) {
+            RM_EMPTY_KRAKEN2(file("${params.outdir}/extracted_reads/kraken2"))
+        }
+        if (params.extract_centrifuge_reads && params.taxid) {
+            RM_EMPTY_CENTRIFUGE(file("${params.outdir}/extracted_reads/centrifuge"))
+        }
+        if (params.extract_diamond_reads && params.taxid) {
+            RM_EMPTY_DIAMOND(file("${params.outdir}/extracted_reads/diamond"))
+        }
+
+        // SUBWORKFLOW: DE NOVO
+
+        // Filter out empty FASTQ files
+        ch_taxid_reads = TAXID_READS.out.reads.filter { meta, reads ->
+            def files = reads instanceof List ? reads: [ reads ] // type checking to ensure that reads is a list
+            files.findAll { filepath ->
+                def file = new java.io.File ( filepath.toString() ) //create file objects
+                file.size() > 0
+            }
+        }
+        // Prepare reads for de-novo assembly
+        ch_denovo_input = ch_taxid_reads.branch { meta, reads ->
+            def files = reads instanceof List ? reads : [reads] // Ensure reads is a list
```

Comment:
what does `instanceof` List do?

Reply:
It checks if reads is a list and if not it converts it into a list. In this context, reads is an array and I need to convert it into a list to check the number of reads in each fastq file.